### PR TITLE
ci: setup iOS app deployment

### PR DESCRIPTION
**Pull request by levibostian**
CI setup to deploy the iOS app on every PR and push.

### CODE REVIEW

File: README.md
Comment (levibostian) [inline README.md:22]: I had better luck getting yarn to work on CI server instead of npm. So, I added `yarn.lock` to the project and use yarn on CI. 

I don't mind yarn or npm for this project. As long as it works on development environments and CI server, I am OK with either. So, this change is up for debate.

File: ios/.gitignore
Comment (levibostian) [inline ios/.gitignore:1]: This is the gitignore we use on iOS SDK. Originally came from [here](https://github.com/github/gitignore/blob/main/Swift.gitignore)

File: README.md
Comment (ami-aman) [inline README.md:34]: Now that we have separate provisioning profile, bundle id & certificates for Ami app, I think we should create a similar document for Ami app to instruct how to setup code signing for it. What do you say ?

File: README.md
Comment (ami-aman) [inline README.md:36]: I think this is a mistake from my end. It should be `...minimum 13.0 in Xcode` since our SDK depends on iOS 13.0 min,

File: ios/Podfile
Comment (ami-aman) [inline ios/Podfile:14]: This is not required. This might result into a conflict. We do not have to add `'CustomerIO/Tracking'` using pods because this is being handled by the our react native package.

Only `CustomerIOMessagingPushAPN` or `CustomerIOMessagingPushFCM` has to be added explicitly.

File: ios/Podfile
Comment (ami-aman) [inline ios/Podfile:28]: Same for Notification Service. https://github.com/customerio/amiapp-reactnative/pull/10/files#r923136545

File: ios/Podfile
Comment (ami-aman) [inline ios/Podfile:48]: This is the right fix for Notification service extension. 👍

File: README.md
Comment (ami-aman) [reply README.md:22]: Either of the both, `npm` or `yarn` are used to install packages. `Yarn` is better than `npm` but `npm` is widely used. I would suggest to add both the possible processes i.e `yarn install` & `npm install` with a one liner explaining each.

I can take this up once this PR is merged.

File: ios/.gitignore
Comment (ami-aman) [reply ios/.gitignore:1]: 👍

File: ios/Podfile
Comment (levibostian) [reply ios/Podfile:48]: Well, it's the right temporary workaround 😉

I have a [bug fix branch](https://github.com/customerio/customerio-ios/tree/levi/compile-notification-service) on the iOS SDK that needs some testing (when I get a moment). Then we don't need this anymore.

File: ios/Podfile
Comment (levibostian) [reply ios/Podfile:14]: Because the Tracking module is a dependency of the messaging push modules, technically the tracking module is always installed alongside the messaging push module. 

Me adding the tracking module to this file is a personal preference. I find it easier to read when child dependencies are listed like this. Helps understand what is and isn't installed.

File: README.md
Comment (levibostian) [reply README.md:34]: Assuming I have time this week, I was going to create a generic code signing document that RH and Ami app can both reference. 

This is a solution that works good enough. The documentation at `https://github.com/customerio/RemoteHabits-iOS/blob/main/docs/dev-notes/DEVELOPMENT.md#setup-code-signing-internal-team-member-only` still works for Ami app. The commands that the document tells you to run are the same in the Ami project so the instructions will still work here.

File: ios/Podfile
Comment (ami-aman) [reply ios/Podfile:14]: I understand that it makes it easier to read but the relevant SDK already exists in the project and is installed as soon as the user does `npm install customerio-reactnative`.

File: ios/Podfile
Comment (ami-aman) [reply ios/Podfile:48]: Awesome. If that gets fixed from the SDK then it will be a great advantage.

**Review by ami-aman — APPROVED** [review]
The PR is good to merge after considering this suggestion https://github.com/customerio/amiapp-reactnative/pull/10#discussion_r923136545

👍

**Pull request merged**
